app/shared/design-system/components/primitives/numeric-gesture-control.ts

```diff
@@ -108,25 +98,72 @@ export const numericGestureControl = (
       }
     }
   };
-  targetNode.style.setProperty("cursor", state.direction);
+
+  handleCursor(targetNode, true);
   eventNames.forEach((eventName) =>
     targetNode.addEventListener(eventName, handleEvent)
   );
   return {
     disconnectedCallback: () => {
-      targetNode.style.removeProperty("cursor");
+      handleCursor(targetNode, false);
       eventNames.forEach((eventName) =>
         targetNode.removeEventListener(eventName, handleEvent)
       );
     },
   };
 };
 
-const cursorNode = new Range().createContextualFragment(`
-  <svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="46" height="15" style="filter:var(--drop-shadow);">
-    <g transform="translate(2 3)">
-      <path d="M 15 4.5L 15 2L 11.5 5.5L 15 9L 15 6.5L 31 6.5L 31 9L 34.5 5.5L 31 2L 31 4.5Z" fill="#111" fill-rule="evenodd" stroke="#FFF" stroke-width="2"></path>
-      <path d="M 15 4.5L 15 2L 11.5 5.5L 15 9L 15 6.5L 31 6.5L 31 9L 34.5 5.5L 31 2L 31 4.5Z" fill="#111" fill-rule="evenodd"></path>
-    </g>
-  </svg>
-`).firstElementChild as HTMLElement;
+const requestPointerLock = (
+  state: State,
+  event: PointerEvent,
+  targetNode: HTMLElement
+) => {
+  targetNode.setPointerCapture(event.pointerId);
+
+  if (shouldUsePointerLock()) {
+    targetNode.requestPointerLock();
+    const cursorNode = new Range().createContextualFragment(`
+      <svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="46" height="15" style="filter:var(--drop-shadow);">
+       <g transform="translate(2 3)">
+         <path d="M 15 4.5L 15 2L 11.5 5.5L 15 9L 15 6.5L 31 6.5L 31 9L 34.5 5.5L 31 2L 31 4.5Z" fill="#111" fill-rule="evenodd" stroke="#FFF" stroke-width="2"></path>
+          <path d="M 15 4.5L 15 2L 11.5 5.5L 15 9L 15 6.5L 31 6.5L 31 9L 34.5 5.5L 31 2L 31 4.5Z" fill="#111" fill-rule="evenodd"></path>
+        </g>
+      </svg>`).firstElementChild as SVGElement;
+    cursorNode.style.position = "absolute";
+    cursorNode.style.left = `${event.offsetX}px`;
+    cursorNode.style.top = `${event.offsetY}px`;
+    cursorNode.style.transform =
+      state.direction === "horizontal" ? "rotate(0deg)" : "rotate(90deg)";
+    cursorNode.style.setProperty(
+      "--drop-shadow",
+      `drop-shadow(${
+        state.direction === "horizontal" ? "0 1px" : "1px 0"
+      } 1.1px rgba(0,0,0,.4))`
+    );
+    state.cursor = cursorNode;
+    targetNode.ownerDocument.documentElement.append(state.cursor as Node);
```

Comment:
is `as` used because it cursor could be undefined/null? If yes then better way is to have an "if" check before you use the variable, because it can actually be unset potentially